python/tests/kat/test_ambassador.py

```diff
@@ -1,47 +1,57 @@
+import pytest
+from conftest import letter_range
+import sys
+
 from kat.harness import Runner, EDGE_STACK
 
 from abstract_tests import AmbassadorTest
 
 # Import all the real tests from other files, to make it easier to pick and choose during development.
 
-import t_basics
-import t_cors
-import t_extauth
-import t_grpc
-import t_grpc_bridge
-import t_grpc_stats
-import t_grpc_web
-import t_gzip
-import t_headerrouting
-import t_hosts
-import t_ip_allow_deny
-import t_loadbalancer
-import t_logservice
-import t_lua_scripts
-import t_mappingtests
-import t_max_req_header_kb
-import t_no_ui
-import t_optiontests
-import t_plain
-import t_ratelimit
-import t_redirect
-#import t_shadow
-#import t_stats
-import t_tcpmapping
-import t_tls
-import t_tracing
-import t_retrypolicy
-import t_consul
-import t_circuitbreaker
-import t_envoy_logs
-import t_ingress
-import t_headerswithunderscoresaction
-import t_listeneridletimeout
-import t_cluster_tag
-import t_queryparameter_routing
-import t_request_header
-import t_regexrewrite_forwarding
-import t_error_response
+if letter_range in ["all","ah"]:
+	import t_basics
+	import t_circuitbreaker
+	import t_cluster_tag
+	import t_consul
+	import t_cors
+	import t_envoy_logs
+	import t_error_response
+	import t_extauth
+	import t_grpc
+	import t_grpc_bridge
+	import t_grpc_stats
+	import t_grpc_web
+	import t_gzip
+	import t_headerrouting
+	import t_headerswithunderscoresaction
+	import t_hosts
+elif letter_range in ["all","ip"]:
+	import t_ingress
+	import t_ip_allow_deny
+	import t_listeneridletimeout
+	import t_loadbalancer
+	import t_logservice
+	import t_lua_scripts
+	import t_mappingtests
+	import t_max_req_header_kb
+	import t_no_ui
+	import t_optiontests
+	import t_plain
```

Comment:
The `plain` thing still worries me. How about comment in here that `plain`, `mappingtests`, and `optiontests` must go together... or, even better, try moving 

```
	import t_mappingtests
	import t_optiontests
```

into `t_plain.py` and let's see what blows up.

Reply:
Will do!